WAS-319 creating a new channel when connecting the AMQPTransport and …

mmartes opened the pull request:
…the old channel is closed

FalconerTC commented:
Would love if we could write an IT to verify the behavior you we're seeing (manually close connection, manually reestablish connection, verify publish works)
https://github.com/RentTheRunway/conduit/blob/master/src/test/java/io/rtr/conduit/AMQPIntegrationIT.java

stephen-hand-RTR reviewed (commented):
Since channels cannot stay open beyond the lifetime of a connection, and the transport only ever has one connection, is there ever a case where a channel is open and not owned by the connection?

I'm just wondering can we not simplify this fix by just checking if the channel exists and is open?

mmartes commented:
@FalconerTC I added an integration test. Good call.

@stephen-hand-RTR You are probably right. I was wary of the channel setter but that is only used in tests. Changed to only check if it is open
